packages/components/src/tab-panel/index.tsx

```diff
@@ -190,12 +196,14 @@ export function TabPanel( {
 					role="tabpanel"
 					id={ `${ selectedId }-view` }
 					className="components-tab-panel__tab-content"
+					ref={ itemRef || undefined }
```

Comment:
This should never be required, only complementary. Useful for managing focus in rendered children.

Reply:
As context, the components crew is working on a new version of the TabPanel component that would support these kinds of features in a more ergonomic API. So ideally, I'd like to make this fix to the Document Overview without adding new APIs to the current TabPanel component (i.e. the `itemRef` prop).

Would it be possible to handle the item ref concerns directly in the consumer (`.edit-post-editor__list-view-container`) without sending it through the TabPanel component?

Reply:
Just to be clear, I mean something [like this](https://github.com/WordPress/gutenberg/commit/2386400cc0b80bd107c96a1e2c47f1db7f7cdd0f). It seems to work but I may not be picking up on all the focus management concerns — are there any requirements that this doesn't cover?

Reply:
Let me give it a test later and see. Thanks for the quick commit @mirka. If it works, it will indeed be simpler than what I was trying to do. I kept getting a rules of hooks violation but I didn't think about using useMergeRefs() earlier.